### PR TITLE
make - fix env BACKENDS detection

### PR DESCRIPTION
**Pull request by jeremylt**
Small bug I made ages ago and just discovered

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -453,8 +453,8 @@ ifneq ($(wildcard $(MAGMA_DIR)/lib/libmagma.*),)
   BACKENDS_MAKE += $(MAGMA_BACKENDS)
 endif
 
-BACKENDS_ENV ?= BACKENDS
-ifneq ($(BACKENDS_ENV),)
+BACKENDS_ENV = $(BACKENDS)
+ifeq ($(BACKENDS_ENV),)
   BACKENDS = $(BACKENDS_MAKE)
   export BACKENDS
 endif
```
Comment (jedbrown) [inline Makefile:460]: What semantic are you looking for in which this isn't sufficient?
```suggestion
BACKENDS ?= $(BACKENDS_MAKE)
```

**Review by jedbrown — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (jeremylt) [reply Makefile:460]: It's really just an artifact of getting the old broken code to work. I think your change would be preferable.

Comment (jeremylt) [reply Makefile]: Woops, we still need to export ttis variable for CI... We should meld the two

File: Makefile [hunk 2]
```diff
@@ -453,11 +453,7 @@ ifneq ($(wildcard $(MAGMA_DIR)/lib/libmagma.*),)
   BACKENDS_MAKE += $(MAGMA_BACKENDS)
 endif
 
-BACKENDS_ENV ?= BACKENDS
-ifneq ($(BACKENDS_ENV),)
-  BACKENDS = $(BACKENDS_MAKE)
-  export BACKENDS
-endif
+BACKENDS ?= $(BACKENDS_MAKE)
```
Comment (jedbrown) [inline Makefile:456]:
```suggestion
BACKENDS ?= $(BACKENDS_MAKE)
export BACKENDS
```

**Pull request merged**